### ISSUE TITLE
Infinite? Loop possible with "improved" relaying logic

### ISSUE THREAD

**Issue by PazerOP** [MihaZupan/HttpToSocks5Proxy#8]
In case it makes any difference: I am using v1.2.0. I am using several different HttpClients in the same process, and each of them are accessed by many threads. Each HttpClient has its own HttpToSocks5Proxy instance. I am using the latest .net core 3 preview (3.0.100 - preview 6).

After a while of running and making many (hundreds) of requests, all available CPU will start being consumed. It appears this is due to an infinite/very large recursive loop in the code added in https://github.com/MihaZupan/HttpToSocks5Proxy/commit/763253437e1fd6788e34285af62dc318f829a358. The call stack looks like:

```
>	System.Net.Sockets.dll!System.Net.Sockets.SocketAsyncEventArgs.DoOperationSendSingleBuffer(System.Net.Sockets.SafeSocketHandle handle, System.Threading.CancellationToken cancellationToken) Line 607	C#
 	System.Net.Sockets.dll!System.Net.Sockets.Socket.SendAsync(System.Net.Sockets.SocketAsyncEventArgs e, System.Threading.CancellationToken cancellationToken) Line 4216	C#
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnReceived()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.StartRelaying()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnSent()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnReceived()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.StartRelaying()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnSent()	Unknown
	[... several hundred lines ...]
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.StartRelaying()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnSent()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnReceived()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.StartRelaying()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnSent()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnReceived()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.StartRelaying()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnSent()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.OnReceived()	Unknown
 	MihaZupan.HttpToSocks5Proxy.dll!MihaZupan.SocketRelay.StartRelaying()	Unknown
 	System.Private.CoreLib.dll!System.Threading.ExecutionContext.RunFromThreadPoolDispatchLoop(System.Threading.Thread threadPoolThread, System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, object state) Line 289	C#
 	System.Private.CoreLib.dll!System.Threading.Tasks.Task.ExecuteWithThreadLocal(ref System.Threading.Tasks.Task currentTaskSlot, System.Threading.Thread threadPoolThread) Line 2406	C#
 	System.Private.CoreLib.dll!System.Threading.ThreadPoolWorkQueue.Dispatch() Line 663	C#
```

I wasn't having any problems with 1.1.3 so I'll probably just downgrade to the time being.

**Comment by MihaZupan** [comment]
Thank you for the report, I've managed to reproduce this behavior.

This is due to an oversight from my side, I'll work on fixing this asap. Sorry for the troubles.

**Comment by MihaZupan** [comment]
@PazerOP Can you see if this is resolved for you with 1.2.1?

**Comment by MihaZupan** [comment]
Closing as resolved

**Comment by PazerOP** [comment]
Sorry about not getting back to you. It appears this issue is fully resolved with 1.2.1.
